primitives/consensus/babe/src/lib.rs

```diff
@@ -356,9 +355,9 @@ pub struct Epoch {
 	/// The epoch index.
 	pub epoch_index: u64,
 	/// The starting slot of the epoch.
-	pub start_slot: SlotNumber,
+	pub start_slot: Slot,
 	/// The duration of this epoch.
-	pub duration: SlotNumber,
+	pub duration: Slot,
```

Comment:
```suggestion
	pub duration: u64,
```

We define `epoch_length` above as `u64` as well.